### DISCUSSION TITLE
gRPC authentication support

### DISCUSSION THREAD

**Discussion by paul35621** [square/wire#3072] [Q&A] [answered]
Hi,

I've created the Protocol Buffers tool [Mouse Melon](https://mousemelon.dev/). Now I want to add gRPC support to it. Initially a gRPC client, but maybe later also a gRPC server.

I like the Wire library for Protocol Buffers and I'm considering to use the Wire library also for gRPC. But, it's important for me that there are no important features missing. For this part of the application I'll be doing the Protocol Buffers serialization myself, so limitations such as no support for groups are not a problem.

Are [these authentication mechanisms](https://grpc.io/docs/guides/auth/) (SSL client certificate / Google token / OAuth token) supported in Wire gRPC? Are there any limitations I should know about?

**Comment by oldergod** [comment, marked as answer]
Hello!

We have not done any work around authentication at the Wire level. If the http client (OkHttp) can't take care of it, then Wire doesn't support it.
